### ISSUE TITLE
TypeScript compiler can't recognize `index.d.ts` due to wrong `package.json`

### ISSUE THREAD

**Issue by psmyrek** [andrewchilds/slack-notify#40]
In the published npm package, there is the `types` field and the `exports` field. But the `types` field is not used if there is an `exports` field. This causes the following problem when trying to import `slack-notify` in TypeScript project:

```
Could not find a declaration file for module 'slack-notify'. '/path/to/node_modules/slack-notify/src/esm/index.mjs' implicitly has an 'any' type.
  There are types at '/path/to/node_modules/slack-notify/src/index.d.ts', but this result could not be resolved when respecting package.json "exports". The 'slack-notify' library may need to update its package.json or typings.ts(7016)
```

This problem indicates that TypeScript's resolution algorithm discovered a JavaScript file, but couldn't find a corresponding type declaration file. The top-level `types` field doesn't act as a fallback resolution for this case.

#### Solution

Copying the `types` field into the `exports` section fixes the problem:

```
"exports": {
  ".": {
    "types": "./src/index.d.ts",
    "require":  "./src/cjs/index.js",
    "default": "./src/esm/index.mjs"
  }
}
```

**Comment by andrewchilds** [comment]
Hi Paweł! Thank you for the report and the proposed solution. I appreciate your help in maintaining this library. I've published v2.0.7 to npm with your change - let me know if that fixes it on your end.

**Comment by psmyrek** [comment]
The problem has been fixed. Thank you for your quick response.
